store/src/sql/access/event.rs

```diff
@@ -61,6 +123,56 @@ impl CeramicOneEvent {
             Err(err) => Err(err.into()),
         }
     }
+
+    /// Splits the evtents into deliverable and non-deliverable events. Non-deliverable events are checked against the database to see if they can be delivered.
+    /// If `require_history` is true, an error is returned if any event is missing history, oherwise they will not be marked as deliverable.
+    async fn with_deliverable_marker<'a>(
+        events: &'a [EventInsertable],
+        pool: &SqlitePool,
+        require_history: bool,
+    ) -> Result<Vec<(bool, &'a EventInsertable)>> {
```

Comment:
yeah, I 100% agree with this. I discussed a bit with Nathaniel and we want to make the an event from the event crate part of the public interface, but some of it comes out of event validation and the migrations stuff he's working on right now.

I made a lot of types to wrap specific behavior (like a group of CIDs rather than a tuple) or a specific database query result but it merits more thought and clean up.